src/braket/pennylane_plugin/braket_device.py

```diff
@@ -675,6 +676,84 @@ def _run_snapshots(self, snapshot_circuits, n_qubits, mapped_wires):
 
         return outcomes
 
+    def _validate_pulse_parameters(self, ev):
+        """Validates pulse input (ParametrizedEvolution) before converting to a PulseGate"""
+
+        # note: the pulse upload on the other side immediately checks that the max amplitude
+        # is not exceeded, so that check has not been included here
+
+        # confirm all frequency and phase values are constant for the duration of a pulse
+        callable_freqs = [pulse.frequency for pulse in ev.H.pulses if callable(pulse.frequency)]
+        callable_phase = [pulse.phase for pulse in ev.H.pulses if callable(pulse.phase)]
+
+        if callable_freqs:
+            raise RuntimeError("Expected all frequencies to be constants but recieved callable(s)")
+        if callable_phase:
+            raise RuntimeError("Expected all phases to be constants but recieved callable(s)")
```

Comment:
We should still allow for constant callable frequency and phase though, right? 😬 
I also just now realized that

Reply:
Ooooh, good point

Reply:
Fixing this now. Current solution does indeed insist that `qml.pulse.constant` is the only allowed "constant callable". I think that's okay, what do you think? 

I think we also need to take this into account in the translation file, right @mudit2812 ?

Reply:
Updated now

Reply:
great! thanks @lillian542 and @mudit2812 👌

Reply:
Going to quickly read over the code now. If the change needed to the translation file is pretty small (which it probably is), I'll just add it to this PR, since Lillian is on vacation now.

Reply:
yeah sounds good, thanks!